Cython/Compiler/Symtab.py

```diff
@@ -896,6 +896,17 @@ def declare_cfunction(self, name, type, pos,
                 elif not in_pxd and entry.defined_in_pxd and type.compatible_signature_with(entry.type):
                     # TODO: check that this was done by a signature optimisation and not a user error.
                     #warning(pos, "Function signature does not match previous declaration", 1)
+
+                    # Cython can't assume anything about cimported functions declared without
+                    # an exception value. This is a performance problem mainly for nogil functions.
+                    if entry.type.nogil and entry.type.exception_value is None and type.exception_value:
+                        performance_hint(
+                            entry.pos,
+                            f"No exception value declared for '{entry.name}' in pxd file.\n"
+                            "Users cimporting this function and calling it without the gil "
+                            f"will always require an exception check.\n"
```

Comment:
extra `f`

Reply:
Doesn't hurt since Python will be happy to ignore it after parsing, but well, yes, it's redundant. I'll remove it in master.